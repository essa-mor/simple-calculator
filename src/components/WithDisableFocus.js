import React from 'react';

function disable(e){
	e.target.blur();
}

function withDisableFocus(Component) {
	return function(props){
		return <Component {...props} onFocus={disable}/>;
	};
}

export default withDisableFocus;