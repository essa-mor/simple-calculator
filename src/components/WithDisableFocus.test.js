import React, { Component } from 'react';
import WithDisableFocus from './WithDisableFocus';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

class MockComponent extends Component {
	render() {
		const { onFocus } = this.props;
		return (<div onFocus={onFocus}>Fake Component</div>);
	}
}

let WithDisableFocusComponent = WithDisableFocus(MockComponent);

describe('WithDisableFocus test', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<WithDisableFocusComponent />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('blur on focus', () => {
		const callback = jest.fn();
		const wrapper = mount(<WithDisableFocusComponent />);
		const target = wrapper.find('div');
		target.simulate('focus', { target: { blur: callback } });
		expect(callback).toHaveBeenCalledTimes(1);
	});
});