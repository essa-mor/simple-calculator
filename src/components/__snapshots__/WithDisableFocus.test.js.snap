// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithDisableFocus test renders correctly 1`] = `
<div
  onFocus={[Function]}
>
  Fake Component
</div>
`;
